### PR TITLE
feat(kibbeh): Custom Emoji Picker

### PR DESCRIPTION
**Pull request by f1shy-dev**
Added a custom emoji picker, and ditched `emoji-mart`! 🎉

**Notes**
- I did have to remove the `:KEKebab:` emote as it wasn't a square, breaking the layout
- I haven't added twemoji yet. Right now it's only custom emotes.

### CODE REVIEW

File: kibbeh/src/modules/room/chat/EmoteData.ts [hunk 1]
```diff
@@ -1574,12 +1556,15 @@ export const customEmojis = [
...
+];
 
 export const emoteMap: Record<string, string> = {};
 
-export type CustomEmote = EmojiData & {
-  customCategory: string;
+export type CustomEmote = {
+  name: string;
+  short_names: Array<string>;
+  keywords: Array<string>;
```
Comment (overlisted) [inline kibbeh/src/modules/room/chat/EmoteData.ts:1566]: why not `string[]`

File: kibbeh/src/modules/room/chat/RoomChatInput.tsx [hunk 2]
```diff
@@ -113,86 +113,61 @@ export const RoomChatInput: React.FC<ChatInputProps> = ({ users }) => {
...
-            const newMsg = [
-              message.slice(0, position),
-              "native" in emoji
-                ? emoji.native
-                : (message.endsWith(" ") ? "" : " ") +
-                  (emoji.colons || "") +
+        <div className={`mb-1`}>
+          <EmojiPicker
+            emojiSet={customEmojis}
+            onEmojiSelect={(emoji) => {
```
Comment (overlisted) [inline kibbeh/src/modules/room/chat/RoomChatInput.tsx:121]:
```suggestion
            onEmojiSelect={emoji => {
```

File: kibbeh/src/modules/room/chat/RoomChatInput.tsx [hunk 3]
```diff
@@ -113,86 +113,61 @@ export const RoomChatInput: React.FC<ChatInputProps> = ({ users }) => {
...
-        <div className="flex-1 mr-2 lg:mr-0 items-end">
+        <div className="flex-1 mr-2 lg:mr-0 items-center bg-primary-700 rounded-8 flex-row">
           <Input
             maxLength={512}
             placeholder={t("modules.roomChat.sendMessage")}
             value={message}
             onChange={(e) => setMessage(e.target.value)}
-            className={`bg-primary-700`}
+            id="room-chat-input"
+            transparent={true}
```
Comment (overlisted) [inline kibbeh/src/modules/room/chat/RoomChatInput.tsx:147]:
```suggestion
            transparent
```

File: kibbeh/src/modules/room/chat/RoomChatInput.tsx [hunk 4]
```diff
@@ -113,86 +113,61 @@ export const RoomChatInput: React.FC<ChatInputProps> = ({ users }) => {
...
+                message.slice(position),
+              ].join("");
+              setMessage(newMsg);
+            }}
+          />
+        </div>
       ) : null}
       <div className="flex items-stretch">
-        <div className="flex-1 mr-2 lg:mr-0 items-end">
+        <div className="flex-1 mr-2 lg:mr-0 items-center bg-primary-700 rounded-8 flex-row">
```
Comment (overlisted) [inline kibbeh/src/modules/room/chat/RoomChatInput.tsx:140]:
```suggestion
        <div className="flex-1 mr-2 lg:mr-0 items-center bg-primary-700 rounded-8">
```
flex-row is set by default

File: kibbeh/src/modules/room/chat/RoomChatInput.tsx [hunk 5]
```diff
@@ -113,86 +113,61 @@ export const RoomChatInput: React.FC<ChatInputProps> = ({ users }) => {
...
               setIsEmoji(false);
               position = 0;
             }}
-            id="room-chat-input"
           />
-          {/* <div
+          <div
             style={{
               color: "rgb(167, 167, 167)",
               display: "flex",
```
Comment (overlisted) [inline kibbeh/src/modules/room/chat/RoomChatInput.tsx:159]: not needed

File: kibbeh/src/modules/room/chat/RoomChatInput.tsx [hunk 6]
```diff
@@ -113,86 +113,61 @@ export const RoomChatInput: React.FC<ChatInputProps> = ({ users }) => {
...
               marginRight: 13,
-              marginTop: -35,
               flexDirection: "row-reverse",
```
Comment (overlisted) [inline kibbeh/src/modules/room/chat/RoomChatInput.tsx:161]: can be set using classname

File: kibbeh/src/ui/EmojiPicker.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { CustomEmote } from "../modules/room/chat/EmoteData";
+
+interface EmojiPickerProps {
+  emojiSet: Array<CustomEmote>;
```
Comment (overlisted) [inline kibbeh/src/ui/EmojiPicker.tsx:5]: why not `CustomEmote[]`

File: kibbeh/src/ui/EmojiPicker.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    >
+      <div
+        className={`grid grid-cols-7 pr-3 gap-2 max-h-16 overflow-y-scroll scrollbar-thin scrollbar-thumb-rounded-xl scrollbar-thumb-primary-800`}
+      >
+        {emojiSet.map((emoji) => (
+          <img
+            key={emoji.name}
+            src={emoji.imageUrl}
+            className={`w-5 max-w-5`}
+            onClick={() => {onEmojiSelect(emoji)}}
```
Comment (overlisted) [inline kibbeh/src/ui/EmojiPicker.tsx:22]:
```suggestion
            onClick={() => onEmojiSelect(emoji)}
```

File: kibbeh/src/ui/EmojiPicker.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+export const EmojiPicker: React.FC<EmojiPickerProps> = ({ emojiSet, onEmojiSelect }) => {
+  return (
+    <div
+      className={`bg-primary-700 rounded-8 flex flex-row flex-grow p-1 max-h-24 pt-2 px-2`}
+    >
+      <div
+        className={`grid grid-cols-7 pr-3 gap-2 max-h-16 overflow-y-scroll scrollbar-thin scrollbar-thumb-rounded-xl scrollbar-thumb-primary-800`}
+      >
+        {emojiSet.map((emoji) => (
```
Comment (overlisted) [inline kibbeh/src/ui/EmojiPicker.tsx:17]:
```suggestion
        {emojiSet.map(emoji => (
```

File: kibbeh/src/ui/Input.tsx [hunk 10]
```diff
@@ -4,11 +4,14 @@ export interface InputProps extends React.ComponentPropsWithoutRef<"input"> {
...
 }
 
 export const Input = forwardRef<HTMLInputElement, InputProps>(
-  ({ className, textarea, error, ...props }, ref) => {
-    const cn = `w-full py-2 px-4 rounded-8 bg-primary-700 text-primary-100 placeholder-primary-300 focus:outline-none ${className} ${
+  ({ className, textarea, error, transparent, ...props }, ref) => {
+    const cn = `w-full py-2 px-4 rounded-8 ${
+      transparent ? `bg-transparent` : `bg-primary-700`
+    } text-primary-100 placeholder-primary-300 focus:outline-none ${className} ${
       error ? `ring-1 ring-secondary` : ``
```
Comment (overlisted) [inline kibbeh/src/ui/Input.tsx:15]: not readable

**Review by overlisted — CHANGES_REQUESTED** [review]
nothing critical you're a good boy

File: kibbeh/src/modules/room/chat/RoomChatInput.tsx (hunk 6 above)
Comment (benawad) [reply kibbeh/src/modules/room/chat/RoomChatInput.tsx]: also instead of using a custom color, use a tailwind color

Comment (f1shy-dev) [reply kibbeh/src/modules/room/chat/RoomChatInput.tsx]: I didn't even write that code it's probably patched in from `kofta`. I'll fix it anyways

**Comment by f1shy-dev** [comment]
Fixed all requested changes.

**Review by overlisted — APPROVED** [review]
haven't tried to run it but looks good

**Pull request merged**